### PR TITLE
Fixes #14446 - Respect STI class in taxonomy search

### PR DESCRIPTION
**Pull request by shlomizadok**
Not sure whether this should go here or in core.
@ares, @ohadlevy thoughts?
Anyhow tested w/ $downstream project and works as expected.

### CODE REVIEW

**Comment by ohadlevy** [comment]
I would open a refactoring issue to enable this usage case, and until its added, will add the workaround here.
However, if its a oneliner change (change from self.name to self.base_name) maybe just fix it in foreman itself? (at least raise a PR to start the discussion).

(in the larger scope - maybe you don't want to use taxonomix at all).

**Comment by ohadlevy** [comment]
Also, I'm missing a test? :)

**Comment by shlomizadok** [comment]
@ohadlevy not needed since https://github.com/theforeman/foreman/pull/3384 ?
image

**Comment by shlomizadok** [comment]
@ohadlevy - Test has been added

**Comment by ares** [comment]
LGTM, sounds as a good fix to core as well, if you open the redmine issue for it, please add a comment with link to this method so it can be tracked and removed later

**Comment by shlomizadok** [comment]
Not so keen to add to core.
Will open an issue on redmine

**Comment by ares** [comment]
> Not so keen to add to core.

why? isn't this broken for Compute Resources now? `Foreman::Model::Ovirt.name` and `Foreman::Model::Ovirt.base_class.name` returns different value

**Comment by shlomizadok** [comment]
I'll re-phrase: Not so keen to add to core by myself.
Opened http://projects.theforeman.org/issues/14458, hopefully someone will pick it

File: app/models/foreman_openscap/arf_report.rb [hunk 1]
```diff
@@ -173,6 +173,17 @@ def destroy
       end
     end
 
+    def self.inner_select(taxonomy, inner_method = which_ancestry_method)
```
Comment (ares) [inline app/models/foreman_openscap/arf_report.rb]: please add a comment "# can be removed after http://projects.theforeman.org/issues/14458 is fixed"

Comment (shlomizadok) [inline app/models/foreman_openscap/arf_report.rb]: Added

**Comment by shlomizadok** [comment]
Closing in favor of https://github.com/theforeman/foreman/pull/3389 which is now merged and was asked to be ported into 1.11

**Pull request closed without merging**